darktable doesn't build on openSUSE with LTO

<!-- IMPORTANT
Bug reports that do not make an effort to help the developers will be closed without notice.
Make sure that this bug has not already been opened and/or closed by searching the issues on GitHub, as duplicate bug reports will be closed.
A bug report simply stating that Darktable crashes is unhelpful, so please fill in most of the items below and provide detailed information.
-->

**Describe the bug**
```
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libdarktable.so.QEG3wD.ltrans43.ltrans.o: in function `icuLikeCompare':
[  453s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:189: undefined reference to `u_foldCase_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:190: undefined reference to `u_foldCase_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libdarktable.so.QEG3wD.ltrans43.ltrans.o: in function `icuRegexpFunc':
[  453s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:67: undefined reference to `u_errorName_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:67: undefined reference to `u_errorName_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:67: undefined reference to `u_errorName_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libdarktable.so.QEG3wD.ltrans43.ltrans.o: in function `icuCaseFunc16':
[  453s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:400: undefined reference to `u_strToUpper_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:67: undefined reference to `u_errorName_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:402: undefined reference to `u_strToLower_67'
[  453s] /usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: /tmp/libdarktable.so.QEG3wD.ltrans43.ltrans.o: in function `icuLoadCollation':
[  453s] /home/abuild/rpmbuild/BUILD/darktable-3.5.0~git102.7a6e50775/src/common/sqliteicu.c:67: undefined reference to `u_errorName_67'
[  453s] collect2: error: ld returned 1 exit status
```

```
$ pkg-config --libs icu-i18n
-licui18n -licuuc -licudata
```

Looking at the linker line I only see /usr/lib64/libicui18n.so but not the other libs ...

Full log:
https://build.opensuse.org/package/live_build_log/graphics:darktable:master/darktable/openSUSE_Tumbleweed/x86_64

This is caused by #7253

--- a/cmake/modules/FindICU.cmake
+++ b/cmake/modules/FindICU.cmake
@@ -12,10 +12,11 @@ include(LibFindMacros)
 libfind_pkg_check_modules(ICU_PKGCONF icu-i18n)
 
 # Find the libraries
-find_library(ICU_LIBRARY
-  NAMES icui18n
-  HINTS ${ICU_PKGCONF_LIBRARY_DIRS}
-)
+foreach(i ${ICU_PKGCONF_LIBRARIES})
+  find_library(_ICU_LIBRARY NAMES ${i} HINTS ${ICU_PKGCONF_LIBRARY_DIRS})
+  LIST(APPEND ICU_LIBRARY ${_ICU_LIBRARY})
+  unset(_ICU_LIBRARY CACHE)
+endforeach(i)
 
 if(ICU_LIBRARY)
   set(ICU_PROCESS_LIBS ${ICU_LIBRARY})

